interfaces/builtin/desktop.go

```diff
@@ -491,6 +491,51 @@ dbus (send)
 /etc/xdg/user-dirs.conf r,
 /etc/xdg/user-dirs.defaults r,
 /run/udev/tags/seat{,/**} r,
+
+# KDE Plasma specific extension
+
+# Used by the KCrash handler
+@{PROC}/sys/kernel/core_pattern r,
+
+# So that KSplash disappears when appropriate
+dbus (receive, send)
+    bus=session
+    path=/KSplash
+    interface=org.kde.KSplash
+    member=setStage
+    peer=(label=unconfined),
+dbus (receive, send)
+    bus=session
+    path=/KSplash
+    interface=org.freedesktop.DBus.Introspectable
+    member=Introspect
+    peer=(label=unconfined),
```

Comment:
In the Core Desktop build I checked, `org.kde.KSplash` is running as `snap.plasma-desktop-session.plasma-ksplash`, so allowing communication with `label=unconfined` doesn't seem necessary.

If the only use of this interface is within the plasma-desktop-session snap, this might not be necessary at all: same-snap D-Bus messages are allowed generally:

https://github.com/canonical/snapd/blob/b0e2fe5580e2e2cc777da55dc414435c5809f528/interfaces/apparmor/template.go#L401-L404

I suspect this might have been necessary with an earlier version of your desktop session snap, but is no longer needed.

Reply:
This is still necessary with the current state of plasma-core22-desktop/plasma-desktop-session. We currently have three unconfined one shot services which are basically just qdbus invocations to those three functions. I preferred to keep them unconfined to avoid duplicating what is effectively tiny pieces of code from upstream.

Obviously if I could I'd like to keep it as is. Let me know what you think.

This applies to this one and your other two comment.

Reply:
Fair enough. The way the rules were written, it looked like it was for communication as a client to an unconfined service rather than allowing unconfined clients to talk to a confined service.